lstchain/scripts/lstchain_create_run_summary.py

```diff
@@ -0,0 +1,192 @@
+the start time of the run, type pf the run: DATA, DRS4, CALI, and
+the reference timestamp and counter of the run.
+"""
+
+import argparse
+import logging
+from pathlib import Path
+
+import numpy as np
+import protozfits
+from astropy.table import Table
+from astropy.time import Time
+from ctapipe.containers import EventType
+from ctapipe.io import EventSource
+from lstchain.paths import parse_r0_filename
+
+# FIXME: take it from somewhere else
+tel_id = 1
+
+
+log = logging.getLogger(__name__)
+
+parser = argparse.ArgumentParser(description="Create run summary file")
+
+parser.add_argument(
+    "-d",
+    "--date",
+    type=str,
+    help="Date for the creation of the run summary in format YYYYMMDD",
+    required=True,
+)
+# TODO: default today()
+
+parser.add_argument(
+    "--r0-path",
+    type=Path,
+    dest="R0_PATH",
+    help="Path to the R0 files. Default is /fefs/aswg/data/real/R0",
+    default=Path("/fefs/aswg/data/real/R0"),
+)
+
+parser.add_argument(
+    "-o",
+    "--output-dir",
+    type=Path,
+    dest="output_dir",
+    help="Directory in which Run Summary file is written",
+    default="./",
+)
+
+
+args = parser.parse_args()
+
+
+def get_list_of_files(date):
+    """Get the list of R0 files from a given date."""
+    path_r0 = args.R0_PATH / date
+    # FIXME: use regular expressions from lstchain.paths.R0_RE
+    list_of_files = path_r0.glob("LST*.fits.fz")
+    return list_of_files
+
+
+def get_list_of_runs(list_of_files):
+    """Get the sorted list of run objects from R0 filenames."""
+    return sorted(parse_r0_filename(file) for file in list_of_files)
+
+
+def get_runs_and_subruns(list_of_run_objects):
+    """
+    Get the list of run numbers and the number of sequenctial files (subruns)
+    of each run for a given date assuming 4 ZFW streams.
+    """
+    NUMBER_OF_STREAMS = 4
+
+    run, number_of_files = np.unique(
+        list(map(lambda x: x.run, list_of_run_objects)), return_counts=True
+    )
+
+    n_subruns = number_of_files / NUMBER_OF_STREAMS
+    return run, n_subruns.astype(int)
+
+
+def type_of_run(run_number, n_events=500):
+    """
+    Get empirically the type of run based on the percentage of
+    pedestals/mono trigger types from the first n_events:
+    100% mono events (trigger 1): DRS4 pedestal run
+    <10% pedestal events (trigger 32): cosmic DATA run
+    ~50% mono, ~50% pedestal events: PEDESTAL-CALIBRATION run
+    First subrun needs to be open.
+    """
+    filename = args.R0_PATH / args.date / f"LST-1.1.Run{run_number:05d}.0000.fits.fz"
+
+    with EventSource(input_url=filename, max_events=n_events) as source:
+        n_pedestal_events = sum(
+            1 for event in source if event.trigger.event_type == EventType.SKY_PEDESTAL
+        )
+        n_sky_events = sum(1 for event in source if event.trigger.event_type == EventType.SUBARRAY)
+
+    # FIXME: Do this classification in some other way?
+    if n_sky_events / n_events > 0.999:
+        run_type = "DRS4"
```

Comment:
In DRS4 runs event type is SUBARRAY? 

This condition would also be true for the case of sky data if, for whatever reason, interleaved events are off.

Reply:
I guess so. Right now DRS4 runs have only mono trigger events. SKY runs have ~1% pedestal trigger and the rest are mono. CALIB is ~50% of each type (mono and pedestal)

See https://www.lst1.iac.es/datacheck/

Reply:
> I guess so. Right now DRS4 runs have only mono trigger events. SKY runs have ~1% pedestal trigger and the rest are mono. CALIB is ~50% of each type (mono and pedestal)
> 
> See https://www.lst1.iac.es/datacheck/

Ok, thx. Anyone knows why DRS4 run events have that type of trigger? Should be pedestals.

Reply:
It has been always like this (at least since Nov 2019)

Reply:
ok... if you were already using this approach to identify run type and it worked, fine for me!